odbc/src/api/functions.rs

```diff
@@ -3499,6 +3500,7 @@ fn sql_tables(
             catalog,
             table,
             table_t,
+            odbc_version != OdbcVersion::Odbc2,
```

Comment:
Yeah that might be better; just thinking about if we want to alias this operation somehow. Also happy to defer this decision to 1689, was just top of mind as I work on it

Reply:
actually a macro works perfectly fine!

Reply:
@jchemburkar 
ah, yes, the type info.... this might get uglier :(
But technically, only type_info.rs needs to be aware of the odbc version in core, no?
For regular columns metadata, the switch can be handle in SQLDescribeCol and SQLColAttribute.
Also, since the odbc version is tied to the environment, you might want to store the info in MongoConnection.
It will break the idea that we avoid ODBC lingo in core, but at this point, it might the easiest for both of you.
More to come for SQL-1689.

Reply:
I went with a macro and put it in handles, since it seemed to fit there best and I wanted to cover it with tests.

Reply:
@nbagnard Yes, it will get a little uglier, but hopefully manageably. I think you're right that type info might be the only one we _need_ to pass it to, but we'll have to be careful in places we use `MongoFields` and others if we don't pass it. I spent some time today trying to figure out the least ugly version. 

And sounds good @terakilobyte , thanks for adding that 😄